skrf/circuit.py

```diff
@@ -297,6 +297,7 @@ def Port(cls, frequency: Frequency, name: str, z0: float = 50) -> Network:
         _media = media.DefinedGammaZ0(frequency, z0=z0)
         port = _media.match(name=name)
         port._is_circuit_port = True
+        port._ext_attrs.append('_is_circuit_port')
```

Comment:
I'm not a fan of manually appending a list after setting an internal value...

What about storing arbitrary information like `_is_circuit_port` in an ordinary dictionary or even an predefined dataclass?

Reply:
Creating an ordinary dictionary for `Network._ext_attrs` is indeed a simple and effective approach. I will update the relevant code accordingly.